c++/include/orc/Type.hh

```diff
@@ -82,6 +82,11 @@ namespace orc {
      * @return a reference to the union type
      */
     virtual Type* addUnionChild(ORC_UNIQUE_PTR<Type> fieldType) = 0;
+
+    /**
+     * Build a Type object from string text representation.
+     */
+    static Type* buildTypeFromString(const std::string& input);
```

Comment:
Let's make it a unique_ptr<Type> to make it clear that the caller owns the pointer.